### PR TITLE
HOUSNAV-48: UI Questions 9.9.9.2

### PR DESCRIPTION
**Pull request by nicholaslyon-dt**
[HOUSNAV-48](https://hous-bssb.atlassian.net/browse/HOUSNAV-48)

## Overview & Purpose
Updates needed to facilitate handling all the questions in section 9.9.9.2 of Walkthrough 9.9.9.

## Summary of Changes
Created PDFDownloadLink component to handle links in content that point the Building Code PDF
Updated string parsing to handle new PDFDownloadLink component
Added GreaterThan logic in NextNavigation

## Testing
Navigate to section 9.9.9.2 (starts with question P11). 

## Notes & Resources
- You can jump to 9.9.9.1 and start there by updated the `startingSectionId` in 9.9.9.json to `9991`.
- Question P15 contains new PDFDownloadLink component
- [Walkthrough 9.9.9 Document](https://docs.google.com/document/d/1VToXTJ10rf__5g771tb5KhCvihJxZPouY5XkYqEnLgk/edit)

## Checklist
- [x] I have written or updated vitests for this work
- [x] I have reviewed the [accessibility guidelines](https://digital.gov.bc.ca/wcag/home/) relevant to this work
- [x] I have reviewed this work with at least one screen reader (when applicable)
- [x] I have added/updated any related documentation

### CODE REVIEW

File: apps/web/utils/string.tsx [hunk 1]
```diff
@@ -53,6 +72,24 @@ export const parseStringToComponents = (
               }
             }
             break;
+          case pdfDownloadLinkName:
+            {
+              const PDFDownloadLinkComponent =
+                customComponents[pdfDownloadLinkName];
+              const props = attributesToProps(domNode.attribs);
+
+              if (PDFDownloadLinkComponent) {
```
Comment (jbook24) [inline apps/web/utils/string.tsx]: What scenario would this be false?

Comment (nicholaslyon-dt) [reply apps/web/utils/string.tsx]: Good call! Not needed anymore since I refactored the type for the `customComponents` object. Removed!

**Comment by jbook24** [comment]
Testing 
- With P(11) being greater than 2 

I am not able to navigate to R(2) as in the screenshot below 
image

**Comment by jbook24** [comment]
Everything looks good from my side as an initial pass that we can iterate on later. 

One point I would like to raise is the warning that is appearing when navigating from question to question. 

**Attached is a snippet** 
image

**Potential Implications** 
When a component changes from uncontrolled to controlled, it means that at some point, the component's state was managed internally (uncontrolled) and then switched to being managed externally via props (controlled). This inconsistency can lead to unpredictable behavior and bugs, as the component might not handle state updates correctly.

**Potential Solution** 
Can we make a position of either "Always Controlled" or "Always Uncontrolled" ?

**Comment by wardm89** [comment]
> Everything looks good from my side as an initial pass that we can iterate on later.
> 
> One point I would like to raise is the warning that is appearing when navigating from question to question.
> 
> **Attached is a snippet** image
> 
> **Potential Implications** When a component changes from uncontrolled to controlled, it means that at some point, the component's state was managed internally (uncontrolled) and then switched to being managed externally via props (controlled). This inconsistency can lead to unpredictable behavior and bugs, as the component might not handle state updates correctly.
> 
> **Potential Solution** Can we make a position of either "Always Controlled" or "Always Uncontrolled" ?

I will work on updating this. It seems to actually be because the radio buttons are set to an undefined state when the question is loaded and the once the selection is made it's state is then controlled. I'm not sure if it's related to Mobx or not but I'll look into it.

**Review by wardm89 — COMMENTED** [review]
LGTM other than the one issue I'll look into and also `type CustomComponentProps` I'm thinking we will want to rethink how we are handling these types so it's cleaner and more sustainable over time. However, I think it's fine for now and works so no need to change anything yet. It can be refactored when the complexity reaches the proverbial "tipping point".

**Review by wardm89 — APPROVED** [review]

**Comment by wardm89** [comment]
Merging since I discussed this with Joshua that it can be merged. 

As a note my change was removing the "defaultValue" for the radio button group since the same functionality is accomplished with "value" when it's passed. Default values are for uncontrolled components so we shouldn't use them in conjunction like this.

**Pull request merged**